### PR TITLE
Fix broken tests and implement build caching

### PR DESCRIPTION
**Pull request by throwawayay**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -67,6 +73,7 @@ jobs:
       run: |
         invoke lint
     - name: Coveralls report
+      continue-on-error: true
```
Comment (introkun) [inline .github/workflows/main.yml:76]: let's keep this

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -15,27 +15,33 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest]
```
Comment (throwawayay) [inline .github/workflows/main.yml:18]: temporary

File: .github/workflows/main.yml (hunk 1 above)
Comment (throwawayay) [reply .github/workflows/main.yml:76]: all `continue-on-error` will be removed before merge 👍

Comment (introkun) [reply .github/workflows/main.yml:76]: thanks

**Comment by introkun** [comment]
Ah.. Coveralls is under maintenance now. https://status.coveralls.io/

**Comment by throwawayay** [comment]
Windows pycurl wheel seems to have the wrong compiled pycurl.
```
E   ImportError: pycurl: libcurl link-time ssl backends (schannel) do not include compile-time ssl backend (openssl)
```

https://github.com/blawar/nut/runs/3641387967#step:10:96

Need to find a different wheel

**Pull request merged**